File: typecast-java/src/main/java/com/neosapience/exceptions/UnauthorizedException.java
package com.neosapience.exceptions;

/**
 * Exception thrown when the API returns a 401 Unauthorized error.
 * 
 * <p>This typically indicates an invalid or missing API key.</p>
 */
public class UnauthorizedException extends TypecastException {
    
    /**
     * Creates a new UnauthorizedException.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     */
    public UnauthorizedException(String message, String responseBody) {
        super(message, 401, responseBody);
    }

    /**
     * Creates a new UnauthorizedException with a cause.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     * @param cause        the underlying cause
     */
    public UnauthorizedException(String message, String responseBody, Throwable cause) {
        super(message, 401, responseBody, cause);
    }
}

File: typecast-php/src/Exceptions/UnauthorizedException.php
<?php

declare(strict_types=1);

namespace Neosapience\Typecast\Exceptions;

class UnauthorizedException extends TypecastException
{
    public function __construct(string $message = 'Unauthorized', ?\Throwable $previous = null)
    {
        parent::__construct($message, 401, $previous);
    }
}
